packages/core/src/utils.ts

```diff
@@ -363,12 +365,8 @@ export function isObservable<T>(value: any): value is Subscribable<T> {
   }
 }
 
-export const symbolObservable: typeof Symbol.observable = (() =>
-  (typeof Symbol === 'function' && Symbol.observable) ||
-  '@@observable')() as any;
-
 export function isStateMachine(value: any): value is AnyStateMachine {
-  return !!value && '__xstatenode' in value;
+  return value instanceof StateMachine;
```

Comment:
the previous version was OK, `instanceof` in the library code has its problems and it's recommended to avoid it. It kinda doesn't matter though as this is going to be removed in https://github.com/statelyai/xstate/pull/3721 . There is no need to introduce a redundant change in this PR so I propose to revert this:
```suggestion
  return !!value && '__xstatenode' in value;
```